include/LightGBM/objective_function.h

```diff
@@ -95,11 +100,13 @@ class ObjectiveFunction {
   LIGHTGBM_EXPORT static ObjectiveFunction* CreateObjectiveFunction(const std::string& str);
 };
 
+namespace Constrained {
```

Comment:
Haven't we split yet the constrained objective function code from the objective_function.h/.cpp into a separate constrained_objective_function.h/.cpp?

I though we already had done that

Failing to put all this new code on separate new files will make maintenance where new patches are merged from LGBM impossible.

Reply:
we did and it was reverted. I've un-reverted it in the last commit :slightly_smiling_face: